backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/callout/C_Campaign_Price.java

```diff
@@ -70,10 +77,47 @@ public void onBPartnerChanged(final I_C_Campaign_Price record)
 		{
 			return;
 		}
-
+		updatePricingInfo(record);
 		record.setC_BP_Group_ID(-1);
 	}
 
+	private static void updatePricingInfo(final I_C_Campaign_Price record)
+	{
+		final BPartnerId bpartnerId = BPartnerId.ofRepoIdOrNull(record.getC_BPartner_ID());
+		if (bpartnerId == null)
```

Comment:
why loading BPartnerId and not checking directly record.getC_BPartner_ID()<=0 then return

Reply:
correction done